metis-media-service/src/main/java/eu/europeana/metis/mediaprocessing/http/AbstractHttpClient.java

```diff
@@ -106,71 +82,153 @@ private void cleanConnections(PoolingHttpClientConnectionManager connectionManag
 
     // Set up the connection.
-    final String resourceUlr = getResourceUrl(resourceEntry);
-    final HttpGet httpGet = new HttpGet(resourceUlr);
-    final HttpClientContext context = HttpClientContext.create();
+    final URI resourceUri = URI.create(getResourceUrl(resourceEntry));
+
+    BodyHandler<InputStream> handler = BodyHandlers.ofInputStream();
+    CancelableBodyWrapper<InputStream> bodyWrapper = new CancelableBodyWrapper<>(handler);
+
+     httpResponse = makeHttpRequest(resourceUri, bodyWrapper);
 
     // Set up the abort trigger
     final TimerTask abortTask = new TimerTask() {
       @Override
       public void run() {
-        LOGGER.info("Aborting request due to time limit: {}.", resourceUlr);
-        httpGet.abort();
+        LOGGER.info("Aborting request due to time limit: {}.", resourceUri.getPath());
+        bodyWrapper.cancel();
+        if (httpResponse.body() != null) {
+          try {
+            httpResponse.body().close();
+          } catch (IOException e) {
+            LOGGER.warn(
+                    "Something went wrong while trying to close the input stream after cancelling the http request.",
+                    e);
+          }
+        }
       }
     };
     final Timer timer = new Timer(true);
     timer.schedule(abortTask, requestTimeout);
 
-    // Execute the request.
-    try (final CloseableHttpResponse responseObject = client.execute(httpGet, context)) {
-
-      // Do first analysis
-      final HttpEntity responseEntity = performThrowingFunction(responseObject, response -> {
-        final int status = response.getCode();
-        if (!httpCallIsSuccessful(status)) {
-          throw new IOException("Download failed of resource " + resourceUlr + ". Status code " +
-                  status + " (message: " + response.getReasonPhrase() + ").");
-        }
-        return response.getEntity();
-      });
+    final int statusCode;
 
-      // Obtain header information.
-      final String mimeType = Optional.ofNullable(responseEntity).map(HttpEntity::getContentType)
-              .orElse(null);
-      final Long fileSize = Optional.ofNullable(responseEntity).map(HttpEntity::getContentLength)
-          .filter(size -> size >= 0).orElse(null);
-      final RedirectLocations redirectUris = context.getRedirectLocations();
-      final URI actualUri = (redirectUris == null || redirectUris.size() == 0) ? httpGet.getUri()
-              : redirectUris.get(redirectUris.size() - 1);
+    if (httpResponse != null) {
+      statusCode = httpResponse.statusCode();
+    } else {
+      throw new IOException("A problem occurred when sending the request");
+    }
 
-      // Process the result.
-      final ContentRetriever content = responseEntity == null ?
-          ContentRetriever.forEmptyContent() : responseEntity::getContent;
-      return createResult(resourceEntry, actualUri, mimeType, fileSize, content);
+    final URI actualUri = httpResponse.uri();
```

Comment:
I think this will not behave the same as before..
Previously if it was a redirect the actualUri is taken after performRedirect method.

Reply:
Aahh you are right, I missed that point, I will put it back to how it was and add a new null check on the `actualUri`